File: backend/src/main/java/com/williamdsw/semsys/domain/dto/ReportDTO.java
package com.williamdsw.semsys.domain.dto;

import java.io.Serializable;
import java.util.Date;
import com.fasterxml.jackson.annotation.JsonFormat;
import com.williamdsw.semsys.domain.Report;

public class ReportDTO implements Serializable
{
	// FIELDS
	
	private static final long serialVersionUID = 1L;
	private Integer id;
	private String title;
	private String content;
	
	@JsonFormat (pattern = "yyyy-MM-dd")
	private Date emission;
	private MeetingScheduleDTO schedule;
	
	// CONSTRUCTORS
	
	public ReportDTO () {}
	public ReportDTO (Report report) 
	{
		super ();
		this.id = report.getId ();
		this.title = report.getTitle ();
		this.content = report.getContent ();
		this.emission = report.getEmission ();
		this.schedule = new MeetingScheduleDTO (report.getSchedule ());
	}
	
	// GETTERS / SETTERS
	
	public Integer getId () 
	{
		return id;
	}
	
	public void setId (Integer id) 
	{
		this.id = id;
	}
	
	public String getTitle () 
	{
		return title;
	}
	
	public void setTitle (String title) 
	{
		this.title = title;
	}
	
	public String getContent () 
	{
		return content;
	}
	
	public void setContent (String content) 
	{
		this.content = content;
	}
	
	public Date getEmission () 
	{
		return emission;
	}
	
	public void setEmission (Date emission) 
	{
		this.emission = emission;
	}
	
	public MeetingScheduleDTO getSchedule () 
	{
		return schedule;
	}
	
	public void setSchedule (MeetingScheduleDTO schedule) 
	{
		this.schedule = schedule;
	}
}

File: frontend/src/app/models/domain/dto/report.dto.ts
import { MeetingScheduleDTO } from './meeting-schedule.dto';

export class ReportDTO {

    private id: number;
    private title: string;
    private content: string;
    private emission: Date;
    private schedule: MeetingScheduleDTO;

    constructor() {}

    public getId(): number {
        return this.id;
    }

    public setId(id: number) {
        this.id = id;
    }

    public getTitle(): string {
        return this.title;
    }

    public setTitle(title: string) {
        this.title = title;
    }

    public getContent(): string {
        return this.content;
    }

    public setContent(content: string) {
        this.content = content;
    }

    public getEmission(): Date {
        return this.emission;
    }

    public setEmission(emission: Date) {
        this.emission = emission;
    }

    public getSchedule(): MeetingScheduleDTO {
        return this.schedule;
    }

    public setSchedule(schedule: MeetingScheduleDTO) {
        this.schedule = schedule;
    }
}
